### ISSUE TITLE
[Demo] Create an example to publish geo datasets on static hosts

### ISSUE THREAD

**Issue by nyurik** [maplibre/maplibre-gl-js#1594]
## Motivation

Various government agencies and organizations often need to publish geodata. Usually organizations need to use paid services to publish them, even though the datasets might be relatively small, or the number of downloads relatively low.  I would like to propose a simple demo that would take all input geojson files and "compile" them into an easy to publish package that can be copy/pasted to a static hosting (like github pages) without any server component.  The output will have:
* a single [pmtiles](https://github.com/protomaps/PMTiles) file with all the source geojson files
* a simple maplibre-gl-js wrapper that shows that pmtile content to the users, loading data directly from the pmtiles (single static file via range requests), rather than using any kind of service
* offer download links for the whole dataset (all geojson files zipped together)
* offer a link to load each individual geojson file (each feature in pmtiles will have a tag that indicates what was the source geojson file for that feature)

## Design Alternatives

Publish data to a paid service

## Design

A big packaging utility to
* convert geojson into pmtiles, where each geojson is a separate feature with a link to the original geojson
* copy all map-showing assets that know which pmtiles to use
* some parameter to (optionally) use some base map

### Mock-Up

```
./publish --base https://example.org --geojson ./ --output ./build
```

### Concepts

publish as part of the maplibre examples, showing the usage of pmtiles, on-click file downloads, ...

### Implementation

i think the utility is better be done in Rust for speed and ease-of-deployment, while the web assets would be using maplibre-gl-js, possibly from some CDN

**Comment by nyurik** [comment]
P.S. I think this would be a good first issue because it does not require any in-depth knowledge of maplibre code, but instead would be a usage example. The whole task is really two separate things: the site and the build utility, and I think it could be done by two independent developers.

**Comment by wipfli** [comment]
That sounds like a useful tool but I think this repo here is not the right place since GL JS is just the web rendering library...

**Comment by wipfli** [comment]
@bdon said at FOSS4G that he is working on a pmtiles output for tippecanoe.

**Comment by bdon** [comment]
tracked here: https://github.com/protomaps/tippecanoe/issues/1

**Comment by krishnaglodha** [comment]
HI, Is this issue still open? I would love to take it and work on it

**Comment by HarelM** [comment]
@krishnaglodha still open. Feel free to hack it!
I would recommend creating an example maybe in the docs repo instead of here...? your call.

**Comment by krishnaglodha** [comment]
Hi @dariokueffer , it slipped through my mind, I'll check repo and issue and ping you in next few hours regarding status. 

Thanks

**Comment by krishnaglodha** [comment]
@dariokueffer / @HarelM   will it be possible to connect for a quick call to understand the issue? I'm trying to create `pmtiles` out of simple json, but facing issue, https://github.com/mapbox/tippecanoe/issues/960

**Comment by HarelM** [comment]
@bdon is the authority related to pmtiles I believe...?

**Comment by bdon** [comment]
Yes, I'd recommend using [felt/tippecanoe](https://github.com/felt/tippecanoe) now as that has direct PMTiles output support.

**Comment by krishnaglodha** [comment]
Hi all , let me know your thoughts on this https://share.vidyard.com/watch/VX75P4QXqkgPnR2UvEekHX

**Comment by HarelM** [comment]
This looks great, but doesn't show the interesting "backend" stuff.
I would recommend two things here:
1. Make a tool to generate pmtiles and/or a tutorial to explain how to use it for the file generation
2. Add a small pmtiles file to the demotiles repo and make sure it's available in the gh pages
3. Add an example in the docs repo that uses the pmtiles plugin + maplibre + the above file. This example should show the address of the demotiles and the pmtiles protocol. The example should be very simple to highlight the simplicity of the usage of the pmtiles.

**Comment by HarelM** [comment]
Moving this to the docs repo.
